### PR TITLE
Merge nltk_tgrep into nltk (#931)

### PR DESCRIPTION
**Pull request by wroberts**
This pull request is a first stab at adding [nltk_tgrep](https://github.com/wroberts/nltk_tgrep) to nltk (Issue #931).

I've rewritten the history of `nltk_tgrep` for this branch in the following ways:
1. move `/nltk_tgrep/tgrep.py` to `/nltk/tgrep.py`
2. move `/nltk_tgrep/tests/test_tgrep.py` to `/nltk/test/unit/test_tgrep.py`
3. trash all other `nltk_tgrep` files

Rewriting history makes the pull request smaller and cleaner, but leaves some `nltk_tgrep` stuff lying around (e.g., issue numbers and old commit hashes in commit messages).

I've also made a few code changes to `nltk_tgrep`:
1. fixed import statements
2. removed MIT license from `tgrep.py`
3. switched from using `future` to `nltk.compat`
4. added `pyparsing` to `/pip-req.txt`

Remaining work to do would be to put what little documentation there is of the module somewhere.  I assume this could go into https://github.com/nltk/nltk/wiki/

For reference (in case I end up doing this again), the shell to make this branch went like this:

``` shell
mkdir nltk
cd nltk
git clone ../nltk_tgrep .
# remove origin
git remote remove origin
# remove all tags
git tag | xargs git tag -d
git filter-branch --tree-filter '
    mkdir -p nltk/test/unit/ ;
    [ -e nltk_tgrep/tgrep.py ] && mv -f nltk_tgrep/tgrep.py nltk/tgrep.py ;
    [ -e tgrep.py ] && mv -f tgrep.py nltk/tgrep.py ;
    [ -e nltk_tgrep/tests/test_tgrep.py ] && mv nltk_tgrep/tests/test_tgrep.py nltk/test/unit/test_tgrep.py ;
    [ -e unittests.py ] && mv -f unittests.py nltk/test/unit/test_tgrep.py ;
    rm -f .travis.yml README.org README.rst __init__.py .gitignore MANIFEST.in ;
    rm -f setup.py nltk_tgrep/VERSION nltk_tgrep/__init__.py nltk_tgrep/tests/__init__.py' --prune-empty -- --all
# rename nltk_tgrep branch from master
git checkout -b nltk_tgrep
git branch -d master
# add github repository for nltk
git remote add origin git@github.com:wroberts/nltk.git
git fetch origin
git checkout master
# update nltk repo to upstream master
git remote add upstream https://github.com/nltk/nltk.git
git fetch upstream
git merge upstream/develop
git push origin master -u
# rename master to develop
git push origin develop -u
# (change github default branch to be develop)
git push origin :master
# merge nltk and nltk_tgrep
git checkout nltk_tgrep
git merge develop
# edit and modify until unit test works
nltk/test/runtests.py -v nltk.test.unit.test_tgrep
# push to github
git push origin nltk_tgrep -u
```

### CODE REVIEW

File: nltk/tgrep.py [hunk 1]
```diff
@@ -0,0 +1,842 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+'''
+TGrep search implementation for NTLK trees.
```
Comment (stevenbird) [inline nltk/tgrep.py]: NLTK!

File: nltk/tgrep.py [hunk 2]
```diff
@@ -0,0 +1,842 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
```
Comment (stevenbird) [inline nltk/tgrep.py:2]: @wroberts are you happy to add our standard header with copyright owned by the NLTK project please?

```
# Natural Language Toolkit: tgrep
#
# Copyright (C) 2001-2015 NLTK Project
# Author: Will Roberts <email>
# URL: <http://nltk.org/>
# For license information, see LICENSE.TXT
```

File: nltk/test/unit/test_tgrep.py [hunk 3]
```diff
@@ -0,0 +1,620 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+'''
+Unit tests for TGrep search implementation for NTLK ParentedTrees.
```
Comment (stevenbird) [inline nltk/test/unit/test_tgrep.py]: Please change to name tgrep in this comment.

**Comment by stevenbird** [comment]
Nice work @wroberts – I'm looking forward to having this in NLTK!

**Comment by wroberts** [comment]
Hi @stevenbird , I've made the changes you've suggested; let me know if there's anything else you'd like changed.  Do you have any thoughts on where to put documentation of the nltk.tgrep code?

**Comment by stevenbird** [comment]
Thanks @wroberts – looks good! Is there more than the tgrep module docstring? That's the most visible location since it will show up in our API docs. You're welcome to add more documentation to `nltk/test/tgrep.doctest` and this will end up on our howto pages.

**Comment by stevenbird** [comment]
@wroberts it would be good if the module docstring contained some examples with their outputs, and if the following lines appeared at the end of the file, so that we automatically tested that the code produces the expected output.

```
if __name__ == "__main__":
    import doctest
    doctest.testmod(optionflags=doctest.NORMALIZE_WHITESPACE)
```

**Comment by wroberts** [comment]
OK, I've put the documentation I have into the module docstring and added the doctest line.  Pull request #961.

**Pull request merged**